src/components/legend.ts

```diff
@@ -45,9 +46,9 @@ export module Components {
       this._colorScale = colorScale;
       this._redrawCallback = (scale) => this.redraw();
       this._colorScale.onUpdate(this._redrawCallback);
-
+      this._formatter = Formatters.identity();
       this.xAlignment("right").yAlignment("top");
-      this.comparator((a: string, b: string) => this._colorScale.domain().indexOf(a) - this._colorScale.domain().indexOf(b));
+      this.comparator((a: string, b: string) => 0);
```

Comment:
The parameters of the old `comparator` are domain names. It fails when `a` and `b` are displayed texts. The new comparator will not sort displayed texts, whose order is the same as the order of the index of domain names.